engines/router/missionctl/log/resultlog/kafka_test.go

```diff
@@ -124,15 +126,22 @@ func TestNewProtobufKafkaLogEntry(t *testing.T) {
 	_, turingLogEntry := makeTestTuringResultLogEntry(t)
 	// Overwrite the turing request id value
 	turingLogEntry.TuringReqId = "testID"
+	// Manually remove the key-value pair corresponding to "Content-Type" in the router and enricher header to prevent
```

Comment:
Could you elaborate more on this non-deterministic behavior? What changes? And why specifically `Content-Encoding` ?

Reply:
It's essentially when the `Header` field of `turingLogEntry`, which is an (unordered) map of strings, get encoded, the order in which the items in `Header` get serialised is not guaranteed to always be the same, leading to some unit tests failing and not others. Since the default test log entry `Header` has 2 keys, one for `Content-Encoding` and the other for `Content-Type`, I simply removed removed one of them to ensure that the output's always the same. 

Another option is to create another test log entry for this kafka test without using `makeTestTuringResultLogEntry`, or to make `makeTestTuringResultLogEntry` use only 1 item for the `Header` field, which would also simplify the other tests using it.

Reply:
I see.. In this case, what do you think about decoding the message using the proto and then using `assert.JSONEq` (as we do in `TestNewJSONKafkaLogEntry`) to compare?

Reply:
Hmm that's possible, but I'm not sure if the intention of the `TestNewProtobufKafkaLogEntry` test was to test the byte encoding of the kafka log entry, as opposed to `TestNewJSONKafkaLogEntry` which seems to be testing for the JSON  output.